hydrus/data/resource_based_classes.py

```diff
@@ -0,0 +1,209 @@
+    :param session: sqlalchemy session
+    :return: dict of object with its properties
+    """
+    type_ = query_info["@type"]
+    id_ = query_info["id_"]
+    database_class = get_database_class(type_)
+    try:
+        object_ = (
+            session.query(database_class)
+            .filter(database_class.id == id_)
+            .one()
+        ).__dict__
+    except NoResultFound:
+        raise InstanceNotFound(type_=type_, id_=id_)
+    # Remove the unnecessary keys from the object retrieved from database
+    object_template = copy.deepcopy(object_)
+    object_template.pop("_sa_instance_state")
+    object_template.pop("id")
+    object_template["@type"] = query_info["@type"]
+    return object_template
+
+
+def delete_object(query_info: Dict[str, str], session: scoped_session) -> None:
+    """
+    Delete the object from the database
+    :param query_info: Dict containing the id and @type of object that has to retrieved
+    :param session: sqlalchemy session
+    """
+    type_ = query_info["@type"]
+    id_ = query_info["id_"]
+    database_class = get_database_class(type_)
+    try:
+        object_ = (
+            session.query(database_class)
+            .filter(database_class.id == id_)
+            .one()
+        )
+    except NoResultFound:
+        raise InstanceNotFound(type_=type_, id_=id_)
+    session.delete(object_)
+    try:
+        session.commit()
+    except Exception:
+        session.rollback()
+
+
+def update_object(
+    object_: Dict[str, Any],
+    query_info: Dict[str, str],
+    session: scoped_session,
+) -> str:
+    """
+    Update the object from the database
+    :param object_: Dict containing updated object properties
+    :param query_info: Dict containing the id and @type of object that has to retrieved
+    :param session: sqlalchemy session
+    :return: the ID of the updated object
+    """
+    # Keep the object as fail safe
+    old_object = get_object(query_info, session)
+    # Delete the old object
+    delete_object(query_info, session)
+    id_ = query_info["id_"]
+    # Try inserting new object
+    try:
+        object_["id"] = id_
+        d = insert_object(object_, session)
+    except Exception as e:
+        # Put old object back
+        old_object["id"] = id_
+        d = insert_object(old_object, session)
+        raise e
+    return id_
+
+
+def get_all_filtered_instances(
+    session: scoped_session, search_params: Dict[str, Any], type_: str
+):
+    """Get all the filtered instances of from the database
+    based on given query parameters.
+    :param session: sqlalchemy scoped session
+    :param search_params: Query parameters
+    :param type_: @type of object to be deleted
+    :return: filtered instances
+    """
+    database_class = get_database_class(type_)
+    try:
+        filtered_instances = (
+            session.query(database_class).filter_by(**search_params).all()
+        )
+    except Exception as e:
```

Comment:
Again don;t use generic exceptions with SQLAlchemy.

Reply:
Yes. Fixed it